### ISSUE TITLE
Maven 3.8.1 requires https repositories

### ISSUE THREAD

**Issue by crummy** [lgrignon/jsweet-maven-plugin#60]
After upgrading to Maven 3.8.1, I get the following error:

```
[ERROR] Failed to execute goal on project dtos: Could not resolve dependencies for project ... Failed to collect dependencies at org.jsweet:jsweet-transpiler:jar:3.0.0 -> org.jsweet.ext:typescript.java-ts.core:jar:2.0.3: Failed to read artifact descriptor for org.jsweet.ext:typescript.java-ts.core:jar:2.0.3: Could not transfer artifact org.jsweet.ext:typescript.java-ts.core:pom:2.0.3 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [jsweet-central (http://repository.jsweet.org/artifactory/libs-release-local, default, releases+snapshots), jsweet-snapshots (http://repository.jsweet.org/artifactory/libs-snapshot-local, default, releases+snapshots), jsweet-external (http://repository.jsweet.org/artifactory/ext-release-local, default, releases+snapshots)] -> [Help 1]
```

I removed all reference to http://repository.jsweet.org/artifactory/libs-release-local in my pom.xml but still see the error - so I assume the link to the HTTP repo is within a dependency POM that I don't control.

Here's the announcement on the https requirement:
https://maven.apache.org/docs/3.8.1/release-notes.html

Maven's documentation suggests using a repository with HTTPS. Is there an equivalent for JSweet?

👋

**Comment by crummy** [comment]
I've been able to build locally by populating my `~/.m2/settings.xml` file like so:

```
<?xml version="1.0" encoding="UTF-8" ?>
<settings xsi:schemaLocation='http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd'
          xmlns='http://maven.apache.org/SETTINGS/1.0.0' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'>
    <mirrors>
        <mirror>
          <id>jsweet-release-http</id>
          <url>http://repository.jsweet.org/artifactory/libs-release-local</url>
          <mirrorOf>external:http:*</mirrorOf>
        </mirror>
        <mirror>
          <id>jsweet-snapshot-http</id>
          <url>http://repository.jsweet.org/artifactory/libs-snapshot-local</url>
          <mirrorOf>external:http:*</mirrorOf>
        </mirror>
        <mirror>
          <id>jsweet-external-http</id>
          <url>http://repository.jsweet.org/artifactory/ext-release-local</url>
          <mirrorOf>external:http:*</mirrorOf>
        </mirror>
        <mirror>
          <id>plugin-jsweet-release-http</id>
          <url>http://repository.jsweet.org/artifactory/plugins-release-local</url>
          <mirrorOf>external:http:*</mirrorOf>
        </mirror>
        <mirror>
          <id>plugin-jsweet-snapshots-http</id>
          <url>http://repository.jsweet.org/artifactory/plugins-snapshot-local</url>
          <mirrorOf>external:http:*</mirrorOf>
        </mirror>
      </mirrors>
</settings>
```

This works in a pinch but ideally wouldn't be necessary.

**Comment by lgrignon** [comment]
Thanks for reporting @crummy  ;)

**Comment by lgrignon** [comment]
JSweet's maven repo is now HTTPS:
https://repository.jsweet.org/artifactory/webapp/#/home

**Comment by davidedc** [comment]
this still happens on my Mac - macOS Big Sur Version 11.2.3, Apple Silicon. The workaround of https://github.com/lgrignon/jsweet-maven-plugin/issues/60#issuecomment-813734467 is still needed.

**Comment by lgrignon** [comment]
@davidedc Do you use https in your JSweet repo URLs? This should work, has been tested many times on different platforms.

**Comment by davidedc** [comment]
> @davidedc Do you use https in your JSweet repo URLs? This should work, has been tested many times on different platforms.

I don't know unfortunately - I'm not familiar with maven and I just used the standard maven installation process and the process of http://www.jsweet.org/getting-started/ . I suppose I might not be "using https in JSweet repo URLs" - but I'm not sure how to check that. I just used the website instead to translate some snippets, so it's OK.

...If prospect users are not meant to be familiar with maven, then it's probably worth checking on a "new" machine what you get when you follow the suggested process...

Thanks for making this software by the way!
